File: python/examples/post_services_cancel_request.py
import pprint
import os
import sys

sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from request import BaseRequest
from dotenv import dotenv_values

config = dotenv_values()

# Example of POST /services/:id/cancel-request

class PostServicesCancelRequest(BaseRequest):
    def __init__(self):
        super().__init__()
    
    def post_cancel(self, service_id):
        return self.make_request('POST', f'/services/{service_id}/cancel-request')

service_id = config['SERVICE_ID'];

request = PostServicesCancelRequest()
response = request.post_cancel(service_id)
pprint.pprint(response)

File: javascript/examples/post_services_cancel_request.js
const baseRequest = require('../base_request.js').baseRequest
const path = require('path');
process.chdir(path.resolve(__dirname, '../../'));
require('dotenv').config();

// Example of POST /services/:service_id/cancel-request

const SERVICE_ID = process.env.SERVICE_ID

const url_path = `/services/${SERVICE_ID}/cancel-request`;
const token = baseRequest.request.token({ path: url_path });
baseRequest.request.post({ path: url_path, token: token }); 